lib/common/rand.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Goro Fuji, Fastly, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include <openssl/rand.h>
+#include <inttypes.h>
+#include "h2o/rand.h"
+#include "h2o/string_.h"
+#include "picotls/openssl.h"
+
+static void format_uuid_rfc4122(char *dst, const uint8_t *uuid)
+{
+    // String Representation:
+    // > UUID  = time-low "-" time-mid "-"
+    // >         time-high-and-version "-"
+    // >         clock-seq-and-reserved
+    // >         clock-seq-low "-" node
+    // See also "4.1.2. Layout and Byte Order" for the layout
+    size_t pos = 0;
```

Comment:
```suggestion
    size_t pos = 0;

```
If we are to add empty lines before `#define` and `#undef`, then I'd say that we'd better have empty lines before them as well.